window/src/os/windows/window.rs

```diff
@@ -390,6 +396,15 @@ fn decorations_to_style(decorations: WindowDecorations) -> u32 {
     }
 }
 
+unsafe fn get_primary_monitor_dpi() -> u32 {
```

Comment:
I'd just remove the `unsafe` from here; you can make it safe by asserting that the resolved `primary` monitor is not null.